contracts/utils/PendingOwnableUpgradeable.sol

```diff
@@ -0,0 +1,166 @@
+        _;
+    }
+
+    /**
+     * @dev Initializes the contract setting `msg.sender` as the initial owner
+     */
+    function __PendingOwnable_init() internal onlyInitializing {
+        __PendingOwnable_init_unchained();
+    }
+
+    function __PendingOwnable_init_unchained() internal onlyInitializing {
+        _transferOwnership(msg.sender);
+    }
+
+    /**
+     * @notice Returns the address of the current owner
+     * @return The address of the current owner
+     */
+    function owner() public view virtual override returns (address) {
+        return _owner;
+    }
+
+    /**
+     * @notice Returns the address of the current pending owner
+     * @return The address of the current pending owner
+     */
+    function pendingOwner() public view virtual override returns (address) {
+        return _pendingOwner;
+    }
+
+    /**
+     * @notice Sets the pending owner address. This address will be able to become
+     * the owner of this contract by calling {becomeOwner}
+     */
+    function setPendingOwner(address pendingOwner_)
+        public
+        virtual
+        override
+        onlyOwner
+    {
+        if (_pendingOwner != address(0))
+            revert PendingOwnable__PendingOwnerAlreadySet();
+        _setPendingOwner(pendingOwner_);
+    }
+
+    /**
+     * @notice Revoke the pending owner address. This address will not be able to
+     * call {becomeOwner} to become the owner anymore.
+     * Can only be called by the owner
+     */
+    function revokePendingOwner() public virtual override onlyOwner {
+        if (_pendingOwner == address(0))
+            revert PendingOwnable__NoPendingOwner();
+        _setPendingOwner(address(0));
+    }
+
+    /**
+     * @notice Transfers the ownership to the new owner (`pendingOwner`).
+     * Can only be called by the pending owner
+     */
+    function becomeOwner() public virtual override onlyPendingOwner {
+        _transferOwnership(msg.sender);
+    }
+
+    /**
+     * @notice Leaves the contract without owner. It will not be possible to call
+     * `onlyOwner` functions anymore. Can only be called by the current owner.
+     *
+     * NOTE: Renouncing ownership will leave the contract without an owner,
+     * thereby removing any functionality that is only available to the owner.
+     */
+    function renounceOwnership() public virtual override onlyOwner {
+        _transferOwnership(address(0));
+    }
+
+    /**
+     * @dev Returns true if this contract implements the interface defined by
+     * `interfaceId`. See the corresponding
+     * https://eips.ethereum.org/EIPS/eip-165#how-interfaces-are-identified[EIP section]
+     * to learn more about how these ids are created.
+     *
+     * This function call must use less than 30 000 gas.
+     */
+    function supportsInterface(bytes4 interfaceId)
+        public
+        pure
+        virtual
+        returns (bool)
+    {
+        return
+            interfaceId == this.supportsInterface.selector ||
```

Comment:
If you use the ERC165Upgradeable file, this should be changed to `super.supportsInterface(interfaceId)`

Reply:
o ok I wouldn't need `interfaceId == this.supportsInterface.selector` anymore then right?